src/aiida/manage/configuration/migrations/migrations.py

```
@@ -498,6 +502,36 @@ def get_oldest_compatible_version(config):
     return config.get('CONFIG_VERSION', {}).get('OLDEST_COMPATIBLE', 0)
 
 
+def config_can_be_downgraded(
+    config: ConfigType,
+    target: Optional[int] = None,
```

Ruff (0.15.21)

[error] Undefined name `Optional`

(F821)